### PR TITLE
Move to .NET 5 RC 2

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Move to .NET 5 RC 2.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "sdk": {
-    "version": "5.0.100-rc.1.20452.10"
+    "version": "5.0.100-rc.2.20480.7"
```
Comment (daxian-dbw) [inline global.json:3]: The officially released RC.2 sdk is `5.0.100-rc.2.20479.15`, see https://dotnet.microsoft.com/download/dotnet/5.0

File: src/Microsoft.PowerShell.Commands.Management/Microsoft.PowerShell.Commands.Management.csproj [hunk 2]
```diff
@@ -47,7 +47,7 @@
 
   <ItemGroup>
     <!-- the following package(s) are from https://github.com/dotnet/corefx -->
-    <PackageReference Include="System.ServiceProcess.ServiceController" Version="5.0.0-rc.1.20451.14" />
+    <PackageReference Include="System.ServiceProcess.ServiceController" Version="5.0.0-rc.2.20479.6" />
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Management/Microsoft.PowerShell.Commands.Management.csproj:50]: The officially released package versions for RC.2 is `5.0.0-rc.2.20475.5`, see https://www.nuget.org/packages/System.ServiceProcess.ServiceController/5.0.0-rc.2.20475.5

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: global.json (hunk 1 above)
Comment (adityapatwardhan) [reply global.json:3]: good catch fixing it.

**Comment by adityapatwardhan** [comment]
@daxian-dbw All fixed, please re-review.

File: src/Microsoft.PowerShell.Commands.Management/Microsoft.PowerShell.Commands.Management.csproj (hunk 2 above)
Comment (doctordns) [reply src/Microsoft.PowerShell.Commands.Management/Microsoft.PowerShell.Commands.Management.csproj]: just a question. Is the plan to have an RC of PowerShell 7.1 based on RTW/RTM .NET 5.0? That is, a release candidate for PowerShell that is built on top of .NET 5.0 and not an RC of 5??

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**